rapids_pytest_benchmark/rapids_pytest_benchmark/rmm_resource_analyzer.py

```diff
@@ -0,0 +1,57 @@
+import os
+import csv
+import rmm
+import tempfile
+
+
+class RMMResourceAnalyzer():
+    """
+    Class to control enabling, disabling, & parsing RMM resource
+    logs.
+    """
+    def __init__(self):
+        self.max_gpu_util = -1
+        self.max_gpu_mem_usage = 0
+        _, self._log_file = tempfile.mkstemp()
+
+    def __del__(self):
+      """
+      Remove log file when instance is garbage collected
+      """
+      os.remove(self._log_file)
+
+    def enable_logging(self):
+      """
+      Enable RMM logging
+      """
+      rmm.enable_logging(log_file_name=self._log_file)
+
+    def disable_logging(self):
+      """
+      Disable RMM logging
+      """
+      rmm.mr._flush_logs()
+      rmm.disable_logging()
+
+
+    def parse_results(self):
+      """
+      Parse CSV results. CSV file has columns:
+      Thread,Time,Action,Pointer,Size,Stream
+      """
+      current_mem_usage = 0
+      # RMM appends the device ID is appended to filename
+      log_file_name = self._log_file + ".dev0"
```

Comment:
I'm wondering if this will break if users are not using GPU 0?  If so, I made a suggestion in the other file about passing the device number in.

Reply:
yeah good point. let me check with Ashwin on this. I pulled this example from their unit test below and meant to check up on this.

https://github.com/rapidsai/rmm/blob/branch-0.19/python/rmm/tests/test_rmm.py#L459-L473

Reply:
from our Slack call:

currently `rapids-pytest-benchmarks` is only being used in a containerized Jenkins job that only has a single GPU available, so this should be fine for our current use case.